php_phongo.c

```diff
@@ -843,7 +845,12 @@ int phongo_execute_command(mongoc_client_t* client, php_phongo_command_type_t ty
 			return false;
 	}
 	if (!result) {
-		phongo_throw_exception_from_bson_error_t(&error TSRMLS_CC);
+		if (error.domain == MONGOC_ERROR_SERVER || error.domain == MONGOC_ERROR_WRITE_CONCERN) {
+			phongo_add_exception_prop(ZEND_STRL("resultDocument"), (zval*) &reply TSRMLS_CC);
+			phongo_throw_exception(PHONGO_ERROR_COMMAND_FAILED TSRMLS_CC, "%s", error.message);
```

Comment:
`manager-executeWriteCommand_error-001.phpt` currently exists to test invalid options. Can you create new 002 and 003 tests to check for a CommandException and dump the result document?

You can consult `manager-executeBulkWrite_error-001.phpt` and `manager-executeBulkWrite_error-003.phpt` for prior art.

I'd suggest using `findAndModify` for both new tests. The `MONGOC_ERROR_WRITE_CONCERN` test can rely on a replica set and specify an outlandish value for `w`. For `MONGOC_ERROR_SERVER`, you can just use an unsupported update operator in the `findAndModify` command, which should trigger a server error.

Reply:
Using `phongo_throw_exception()` here is problematic, as it always sets the exception code to 0. This actually causes quite a bit of the test suite to be skipped because [`CLEANUP()`](https://github.com/mongodb/mongo-php-driver/blob/0a745ee8dee43ca7952afea6217678302e13b5bc/tests/utils/tools.php#L239) can no longer catch specific server-side exception codes when attempting to drop a nonexistent collection.

Now, we also can't use `phongo_throw_exception_from_bson_error_t()` here because all that knows is that the `bson_error_t`'s domain is either `MONGOC_ERROR_SERVER` or `MONGOC_ERROR_WRITE_CONCERN`, which would return a ServerException or RuntimeException, respectively.

Since we know the exact exception class we'd like to use here, and we have the error message and code (from `bson_error_t`), let's call `zend_throw_exception()` directly:

```c
zend_throw_exception(php_phongo_commandexception_ce, error.message, error.code TSRMLS_CC);
```

Note: `phongo_exception_from_mongoc_domain()` should probably be fixed to return a ServerException for the `MONGOC_ERROR_WRITE_CONCERN` domain, since libmongoc [documents](http://mongoc.org/libmongoc/current/errors.html) that its error code always originates from the server. I've opened [PHPC-1143](https://jira.mongodb.org/browse/PHPC-1143) to track that.